playbooks/roles/swag_ember/tasks/main.yml

```diff
@@ -0,0 +1,66 @@
+---
+- name: Create swag_ember repo
+  file:
+    path={{swag_ember__root}}
+    owner={{swag_ember__owner}}
+    group={{swag_ember__group}}
+    mode=2775
+    state=directory
+
+- name: Clone swag_ember repo
+  sudo: yes
+  sudo_user: "{{swag_ember__owner}}"
+  git:
+    repo=https://github.com/tunapanda/swag-ember
+    recursive=true
+    dest={{swag_ember__root}}
+    update=false
+
+- name: Deploy swag_ember config
+  template:
+    src="environment.js"
+    dest="{{swag_ember__root}}/config/environment.js"
+    owner="{{swag_ember__owner}}"
+    group="{{swag_ember__group}}"
+
+- name: Create a swap file
+  sudo: yes
+  command: fallocate -l 1G /swapfile
+  ignore_errors: true
+
+- name: Swap permission
+  sudo: yes
+  file:
+    path=/swapfile
+    mode=0600
+    state=file
+
+- name: make Swap
+  sudo: yes
+  command: mkswap /swapfile
+  ignore_errors: true
+
+- name: Enable Swap
+  sudo: yes
+  command: swapon /swapfile
+  ignore_errors: true
+
```

Comment:
There's actually a script you can call to do all this in [scripts/create_swapfile.sh](https://github.com/tunapanda/provision/blob/master/scripts/create_swapfile.sh). It checks for existing swap space and creates as much as is needed to get swap = RAM*2.

It's currently run manually as part of the setup process for a cubietruck deployment, but you should be able to do something like this:

 ```
- name: Create swap
  command: "{{provision__base_dir}}/scripts/create_swapfile.sh"
```